broker/src/main/java/org/apache/rocketmq/broker/controller/ReplicasManager.java

```diff
@@ -268,6 +268,7 @@ public void changeToMaster(final int newMasterEpoch, final int syncStateSetEpoch
 
                 // Notify ha service, change to master
                 this.haService.changeToMaster(newMasterEpoch);
+                this.haService.setBrokerControllerId(this.brokerControllerId);
```

Comment:
After registration is completed, the brokerControllerId is determined, and there is no need to set it every time the master is switched.

Reply:
OK, I'll modify this logic.